src/middlewared/middlewared/plugins/cloud/path.py

```diff
@@ -0,0 +1,26 @@
+import os
+
+from middlewared.service import CallError
+
+
+def get_remote_path(provider, attributes):
+    remote_path = attributes["folder"].rstrip("/")
+    if not remote_path:
+        remote_path = "/"
+    if provider.buckets:
+        remote_path = f"{attributes['bucket']}/{remote_path.lstrip('/')}"
+    return remote_path
+
+
+async def check_local_path(middleware, path, *, check_mountpoint=True, error_text_path=None):
+    error_text_path = error_text_path or path
+
+    if not await middleware.run_in_thread(os.path.exists, path):
```

Comment:
Instead of spinning up 2 new threads each time this is called, we can do something like
```
try:
    info = await middleware.run_in_thread(os.stat, path)
except FileNotFoundError:
    raise CallError(f"Directory {error_text_path!r} does not exist")
else:
    if not: stat.S_ISDIR(info.st_mode):
        raise CallError(f"{error_text_path!r} is not a directory")

Reply:
@yocalebo it does not spin new threads

Reply:
Yeah, I used bad verbiage. Can we still consider taking my approach? It still only uses 1 thread in the thread-pool instead of 2.